### PR TITLE
Feature/ui client oauth2 implicit grant

### PR DESCRIPTION
**Pull request by krzpiesiewicz**
A feature that enables implicit grant.

### CODE REVIEW

File: ui/client/containers/NussknackerInitializer.tsx [hunk 1]
```diff
@@ -113,29 +115,39 @@ class NussknackerInitializer extends React.Component<Props, State> {
...
+          return Promise.resolve(response)
+        }).catch(error => {
+          this.setState({error: this.state.errors[NussknackerInitializer.ACCESS_TOKEN_CODE]})
+          return Promise.reject(error)
+        }).finally(() => {
+          this.props.history.replace({search: null})
+        })
+      }
+
+      console.log(settings)
```
Comment (lciolecki) [inline ui/client/containers/NussknackerInitializer.tsx:138]: No no - please remove this debug stuff ;)

File: ui/client/package.json [hunk 2]
```diff
@@ -17,59 +17,61 @@
     "test": "./node_modules/karma/bin/karma start --single-run"
```
Comment (lciolecki) [inline ui/client/package.json:17]: Do you realy need update these all of packages? I think update shpuld be done in other PR.

Comment (krzpiesiewicz) [reply ui/client/package.json]: I have restored the old package.json and package-lock.json files. You should see a rebased version.

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Configuration.scala [hunk 3]
```diff
@@ -14,6 +15,7 @@ case class OAuth2Configuration(method: AuthenticationMethod,
                                profileUri: URI,
                                accessTokenUri: URI,
                                redirectUri: URI,
+                               override val implicitGrantEnabled: Boolean,
```
Comment (lciolecki) [inline engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Configuration.scala:18]: You don't need override val.

File: ui/client/containers/NussknackerInitializer.tsx [hunk 4]
```diff
@@ -38,8 +38,8 @@ class NussknackerInitializer extends React.Component<Props, State> {
   public static HTTP_APPLICATION_CODE = 500
   public static ACCESS_TOKEN_CODE = 1024
 
-  redirectToAuthorizeUrl = () => {
-    window.location.replace(this.props.authenticationSettings.authorizeUrl)
+  redirectToAuthorizeUrl(settings) {
```
Comment (lciolecki) [inline ui/client/containers/NussknackerInitializer.tsx:41]: It's tsx file so, you can add type for settigns.

Comment (krzpiesiewicz) [reply ui/client/containers/NussknackerInitializer.tsx:41]: `authenticationSettings` has no type yet:
```
export type SettingsState = {
  loggedUser: Partial<User>,
  featuresSettings: $TodoType,
  authenticationSettings: $TodoType,
  analyticsSettings: $TodoType,
  processDefinitionData: ProcessDefinitionData,
  availableQueryableStates: $TodoType,
}
```
and
```
declare type $TodoType = any
```

Comment (lciolecki) [reply ui/client/containers/NussknackerInitializer.tsx:41]: But you can create authenticationSettings? You know what DTO is on backend :)

**Review by lciolecki — APPROVED** [review]

**Pull request merged**